app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SessionDetailViewModel.java

```diff
@@ -139,10 +143,12 @@ public void onClickFab(@SuppressWarnings("unused") View view) {
             mySessionsRepository.delete(session)
                     .subscribe((result) -> Log.d(TAG, "Deleted my session"),
                             throwable -> Log.e(TAG, "Failed to delete my session", throwable));
+            unregisterAlarm();
         } else {
             mySessionsRepository.save(session)
                     .subscribe(() -> Log.d(TAG, "Saved my session"),
                             throwable -> Log.e(TAG, "Failed to save my session", throwable));
+            registerAlarm();
```

Comment:
If the time is within 10 mins to start the session, the alerm should not be registered.
Can you check like below?

```java
public boolean shouldNotify(Session session, long remindDuration) {
    Date now = new Date();
    long diff = session.stime.getTime() - now.getTime();
    return remindDuration < diff;
}
```